Tests/TestingTests/AttachmentTests.swift

```diff
@@ -424,6 +431,53 @@ struct AttachmentTests {
       try attachment.attachableValue.withUnsafeBufferPointer(for: attachment) { _ in }
     }
   }
+
+  @Test("Attach Codable-conformant type with metadata")
+  func codableWithMetadata() async throws {
+    let attachableValue = MyCodableAttachable(string: "abc123")
+    let attachment = Attachment(
+      attachableValue,
+      metadata: .init(
+        format: .propertyListFormat(.xml)
```

Comment:
Could you add a similar test that checks the happy path of .json metadata serialization